### PR TITLE
Noimage

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Part of #5643 

Moves "noimage" images (which are used when a product has no image uploaded yet) under /public so they can be accessed easily in non-digest form.

#### What should we test?

Images for products and variants that are shown when they have no uploaded image (the default OFN logo) should be working in the shopfront, cart dropdown, cart page, and admin products page.

#### Release notes

Adjusted default product image paths for Rails 4.

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
Working as intended on UK staging.

File: app/views/spree/admin/variants/_autocomplete.js.erb [hunk 1]
```diff
@@ -4,7 +4,7 @@
       {{#if variant.image }}
         <img src='{{variant.image}}' />
       {{ else }}
-        <img src='<%= image_path("noimage/mini.png") %>' />
+        <img src='/noimage/mini.png' />
```
Comment (sauloperez) [inline app/views/spree/admin/variants/_autocomplete.js.erb:7]: but don't we need Rails url helpers so the fingerprinted URL is picked up?

Comment (Matt-Yorkley) [reply app/views/spree/admin/variants/_autocomplete.js.erb:7]: I made a slightly different choice for this specific case. These default images are used in various API endpoints, serializers and also in javascript files where generating the fingerprint is a bit messy. Moving them under /public means we can use the non-fingerprinted versions of these images (which is what we were doing before, anyway).

What do you think?

Comment (sauloperez) [reply app/views/spree/admin/variants/_autocomplete.js.erb:7]: I see. That's good enough

**Comment by filipefurtad0** [comment]
Hey @Matt-Yorkley ,

I decided to stage this in another server, I took staging-FR.

Checked that the logo is there when no pic is associated with that item on:

- Shopfront
image


- Cart-dropdow

image

- Admin Products Page

image

- Cart Page

The logo seems to be missing here:

Staging-FR
image

Indeed in staging UK it looks fine:

Staging-UK
image

I'm re-staging this PR in UK and post again in a few moments.

**Comment by Matt-Yorkley** [comment]
Yes, there's a bit missing. Thanks for spotting it!

I'm adding another commit now that should fix it :+1:

**Comment by filipefurtad0** [comment]
Just checked in UK, after re-staging:
image

**Comment by Matt-Yorkley** [comment]
@filipefurtad0 the latest build should be good :+1:

@sauloperez and @luisramos0 I've added 2 extra commits here :point_up:

**Comment by filipefurtad0** [comment]
All good now @Matt-Yorkley ,

image

Good as well on the previous spots on the app:
image

image

Good to go!

**Comment by filipefurtad0** [comment]
Short note:
I realize now I skipped the code review step before testing moving to ready to go. I hope this is ok! Ping @luisramos0 @sauloperez

File: app/views/spree/orders/_bought.html.haml [hunk 2]
```diff
@@ -16,10 +16,10 @@
       %td.cart-item-description
 
         %div.item-thumb-image
-          - if variant.images.length == 0
-            = mini_image(variant.product)
+          - if variant.product.images.length == 0
+            = image_tag("/noimage/mini.png")
```
Comment (luisramos0) [inline app/views/spree/orders/_bought.html.haml]: AFAIK the current status re variant images is that variant specific images can be added to the system in the product images page BUT they are not used anywhere. SO, we can make all our code go to variant.product.images ONLY. Which is what you are doing here. So I think this is ok, just sharing my thoughts :-)

**Comment by filipefurtad0** [comment]
Hi again @luisramos0 @Matt-Yorkley ,

I've found something which may be related to this PR, on the admin orders page /admin/orders/Rxxxxx/edit:

image

While testing I'm pretty sure I did not test this, didn't think of it :-/
Opening an issue and linking to this one.

**Pull request merged**